### PR TITLE
haskellPackages.greenclip: unbreak

### PR DESCRIPTION
**Pull request by teto**
###### Motivation for this change
adding xscrnsaver that is a new transitive dep of greenclip.

Maybe I should remove the broken status somewhere else, but maybe it will be
fixed automatically on next update ?

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 1]
```diff
@@ -275,6 +275,7 @@ self: super: {
   github-rest = dontCheck super.github-rest;  # test suite needs the network
   gitlib-cmdline = dontCheck super.gitlib-cmdline;
   GLFW-b = dontCheck super.GLFW-b;                      # https://github.com/bsl/GLFW-b/issues/50
+  greenclip = markUnbroken (addExtraLibrary super.greenclip pkgs.xlibs.libXScrnSaver);
```
Comment (cdepillabout) [inline pkgs/development/haskell-modules/configuration-common.nix:278]: Could you instead remove `greenclip` from the file `pkgs/development/haskell-modules/configuration-hackage2nix/broken.yaml`?

Also, do you know why this extra library is needed?  If it should actually be added to the cabal file, could you create an issue upstream?  Or if this is a library that `cabal2nix` doesn't know about, could you send a PR there?

Comment (teto) [reply pkgs/development/haskell-modules/configuration-common.nix:278]: this seems like a valid dependency https://github.com/erebe/greenclip/issues/110. I've submitted https://github.com/NixOS/cabal2nix/pull/496 : when this last PR is merged, when can I expect that version of cabal2nix to be usable from nixpkgs ?

Comment (maralorn) [reply pkgs/development/haskell-modules/configuration-common.nix:278]: Immediately after someone runs the "update-cabal2nix-unstable" maintainer script.

Comment (teto) [reply pkgs/development/haskell-modules/configuration-common.nix:278]: could I do that in this PR too ? or is it too impactful ?

Comment (maralorn) [reply pkgs/development/haskell-modules/configuration-common.nix:278]: I see no reason against it.

**Comment by cdepillabout** [comment]
I merged in NixOS/cabal2nix#496, updated our `cabal2nix-unstable` version, umarked `greenclip` as broken in cd36dc4d98c6438, and regenerated the hackage packages in 99e97c88f0e.

@teto If you're a `greenclip` user and you want to keep it working, I urge you to become a maintainer for `greenclip` by adding yourself to https://github.com/NixOS/nixpkgs/blob/259177f1097be0f8de5969242e98f8e026df8037/pkgs/development/haskell-modules/configuration-hackage2nix/main.yaml#L123.  You'll get pinged if greenclip ever stops building on Hydra.  For instance, in a message like https://github.com/NixOS/nixpkgs/pull/123682#issuecomment-847454060.

**Comment by teto** [comment]
@cdepillabout I have a bad experience with automatic notifcations, just too much spam.
I would like to switch to wayland setup soon so I may ditch greenclip as well. Until then I mightfix issues in greenclip if noone beat me to hit. Thanks for your help solving this issue and your overall work on the haskell ecossytem.

**Pull request closed without merging**